UM-USElab/gradecraft-development
# 3686 Authenticate through Google

### Status
**IN DEVELOPMENT**

### Description
As an instructor, I would like to be able to declare that I and my class would like to use Google Authentication to access GradeCraft

### Related PRs
N/A

### Todos

### Deploy Notes
N/A

### Gem dependencies
N/A

### Migrations
NO

### Steps to Test or Reproduce
As an instructor, create a new learner. That new Learner should receive an email.

From the Learner's email, click the activate account link. This will take the Learner to the set password page. Instead of setting the password, use the Sign in with Google button. This should create a Google specific authorization for the new Learner and successfully log the Learner in to GradeCraft.

If successfully logged into GradeCraft, log out. Now, when the Learner is taken back to the standard login page, the Learner can press the "Sign in with Google" button and be logged into GradeCraft.

If I have never authenticated through Google, then I should be taken to the Authorization error page. This path will be modified in a later edition to instead offer to create an account via Google.

### Impacted Areas in Application
* Sign In

======================
Closes #3686
